src/Uno.UI/UI/Xaml/CornerRadius.cs

```diff
@@ -4,91 +4,186 @@
+	{
+		TopLeft = topLeft;
+		TopRight = topRight;
+		BottomLeft = bottomLeft;
+		BottomRight = bottomRight;
+	}
 
-		/// <summary>Gets or sets the radius of the bottom-right corner. </summary>
-		/// <returns>The radius of the bottom-right corner. The default is 0.</returns>
-		public double BottomRight { get; set; }
+	private static bool Equals(CornerRadius left, CornerRadius right)
+		=> left.TopLeft == right.TopLeft
+			&& left.TopRight == right.TopRight
+			&& left.BottomLeft == right.BottomLeft
+			&& left.BottomRight == right.BottomRight;
+
+	/// <inheritdoc />
+	public bool Equals(CornerRadius other)
+		=> Equals(this, other);
+
+	/// <inheritdoc />
+	public override bool Equals(object obj)
+		=> obj is CornerRadius other && Equals(this, other);
+
+	/// <inheritdoc />
+	public override int GetHashCode()
+		=> TopLeft.GetHashCode()
+			^ TopRight.GetHashCode()
+			^ BottomLeft.GetHashCode()
+			^ BottomRight.GetHashCode();
+
+	/// <inheritdoc />
+	public override string ToString()
+		=> "TopLeft: {0}, TopRight: {1}, BottomRight: {2}, BottomLeft: {3}".InvariantCultureFormat(TopLeft, TopRight, BottomRight, BottomLeft);
+
+	internal string ToStringCompact()
+		=> string.Format(CultureInfo.InvariantCulture, "[CornerRadius: {0}-{1}-{2}-{3}]", TopLeft, TopRight, BottomRight, BottomLeft);
+
+	/// <summary>
+	/// Provides a Zero-valued corner radius.
+	/// </summary>
+	public static readonly CornerRadius None = new CornerRadius(0);
+
+	/// <summary>
+	/// Builds a uniform radius from a double;
+	/// </summary>
+	public static implicit operator CornerRadius(double uniformRadius) => new CornerRadius(uniformRadius);
+
+	/// <summary>
+	/// Determines if two CornerRadius instances are equal.
+	/// </summary>
+	public static bool operator ==(CornerRadius cr1, CornerRadius cr2) => Equals(cr1, cr2);
+
+	/// <summary>
+	/// Determines if two CornerRadius instances are equal.
+	/// </summary>
+	public static bool operator !=(CornerRadius cr1, CornerRadius cr2) => !Equals(cr1, cr2);
+
+	/// <summary>
+	/// Retrieves the radii for a border.
+	/// </summary>
+	/// <param name="borderThickness">Border thickness.</param>
+	/// <param name="outer">True to return outer corner radii, false for inner.</param>
+	/// <returns>Radii.</returns>
+	internal FullCornerRadius GetRadii(Size elementSize, Thickness borderThickness, bool outer)
+	{
+		var halfLeft = borderThickness.Left * 0.5;
+		var halfTop = borderThickness.Top * 0.5;
+		var halfRight = borderThickness.Right * 0.5;
+		var halfBottom = borderThickness.Bottom * 0.5;
 
-		/// <summary>Gets or sets the radius of the bottom-left corner. </summary>
-		/// <returns>The radius of the bottom-left corner. The default is 0.</returns>
-		public double BottomLeft { get; set; }
+		double leftTop, topLeft, topRight, rightTop, rightBottom, bottomRight, leftBottom, bottomLeft;
+		leftTop = topLeft = topRight = rightTop = rightBottom = bottomRight = leftBottom = bottomLeft = 0;
```

Comment:
I think it would be best to differentiate the local variables something else but the casing.